providers/statsig/pkg/provider.go

```diff
@@ -0,0 +1,436 @@
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+
+	return of.InterfaceResolutionDetail{
+		Value: value,
+		ProviderResolutionDetail: of.ProviderResolutionDetail{
+			FlagMetadata: flagMetadata,
+		},
+	}
+}
+
+func toStatsigUser(evalCtx of.FlattenedContext) (*statsig.User, error) {
+	if len(evalCtx) == 0 {
+		return &statsig.User{}, nil
+	}
+
+	statsigUser := statsig.User{}
+	for key, origVal := range evalCtx {
+		switch key {
+		case "UserID":
+			val, ok := toStr(origVal)
+			if !ok {
+				return nil, fmt.Errorf("key `%s` can not be converted to string", key)
+			}
+			statsigUser.UserID = val
+		case "Email":
+			val, ok := toStr(origVal)
+			if !ok {
+				return nil, fmt.Errorf("key `%s` can not be converted to string", key)
+			}
+			statsigUser.Email = val
+		case "IpAddress":
+			val, ok := toStr(origVal)
+			if !ok {
+				return nil, fmt.Errorf("key `%s` can not be converted to string", key)
+			}
+			statsigUser.IpAddress = val
+		case "UserAgent":
+			val, ok := toStr(origVal)
+			if !ok {
+				return nil, fmt.Errorf("key `%s` can not be converted to string", key)
+			}
+			statsigUser.UserAgent = val
+		case "Country":
+			val, ok := toStr(origVal)
+			if !ok {
+				return nil, fmt.Errorf("key `%s` can not be converted to string", key)
+			}
+			statsigUser.Country = val
+		case "Locale":
+			val, ok := toStr(origVal)
+			if !ok {
+				return nil, fmt.Errorf("key `%s` can not be converted to string", key)
+			}
+			statsigUser.Locale = val
+		case "AppVersion":
+			val, ok := toStr(origVal)
+			if !ok {
+				return nil, fmt.Errorf("key `%s` can not be converted to string", key)
+			}
+			statsigUser.AppVersion = val
+		case "Custom":
+			if val, ok := origVal.(map[string]interface{}); ok {
+				statsigUser.Custom = val
+			} else {
+				return nil, fmt.Errorf("key `%s` can not be converted to map", key)
+			}
+		case "PrivateAttributes":
+			if val, ok := origVal.(map[string]interface{}); ok {
+				statsigUser.PrivateAttributes = val
+			} else {
+				return nil, fmt.Errorf("key `%s` can not be converted to map", key)
+			}
+		case "StatsigEnvironment":
+			if val, ok := origVal.(map[string]string); ok {
+				statsigUser.StatsigEnvironment = val
+			} else {
+				return nil, fmt.Errorf("key `%s` can not be converted to map", key)
+			}
+		case "CustomIDs":
+			if val, ok := origVal.(map[string]string); ok {
+				statsigUser.CustomIDs = val
+			} else {
+				return nil, fmt.Errorf("key `%s` can not be converted to map", key)
+			}
+		case featureConfigKey:
+		default:
+			return nil, fmt.Errorf("key `%s` is not mapped", key)
+		}
+	}
+
+	return &statsigUser, nil
+}
+
+func toStr(val interface{}) (string, bool) {
+	switch v := val.(type) {
+	case string:
+		return v, true
+	case int, int8, int16, int32, int64:
+		return fmt.Sprintf("%d", v), true
+	case float32, float64:
+		return fmt.Sprintf("%.6f", v), true
+	case bool:
+		return fmt.Sprintf("%t", v), true
+	default:
+		return "", false
+	}
+}
+
+type FeatureConfigType string
```

Comment:
Seems types defined beyond this point are internal only, so use the lowercase first letter to avoid exporting

Reply:
Using it at tests, and similarly the consumer can use it, no ?

Reply:
Noticed that you use it in the example, where it's a part of the context 👍